CondCore/EcalPlugins/plugins/EcalChannelStatus_PayloadInspector.cc

```diff
@@ -0,0 +1,109 @@
+#include "CondCore/Utilities/interface/PayloadInspectorModule.h"
+#include "CondCore/Utilities/interface/PayloadInspector.h"
+#include "CondCore/CondDB/interface/Time.h"
+
+// the data format of the condition to be inspected
+#include "CondFormats/EcalObjects/interface/EcalChannelStatus.h"
+
+#include <memory>
+#include <sstream>
+
+
+
+namespace {
+
+  /*******************************************************
+   
+     2d histogram of ECAL barrel channel status of 1 IOV 
+
+  *******************************************************/
+
+
+  // inherit from one of the predefined plot class: Histogram2D
+  class EcalChannelStatusEBMap : public cond::payloadInspector::Histogram2D<EcalChannelStatus> {
+
+  public:
+    EcalChannelStatusEBMap() : cond::payloadInspector::Histogram2D<EcalChannelStatus>( "ECAL Barrel channel status - map ",
+										       "iphi", 360, 1., 361., "ieta", 171, -85., 86.) {
+     Base::setSingleIov( true );
+    }
+
+    // Histogram2D::fill (virtual) needs be overridden - the implementation should use fillWithValue
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+
+      for( auto iov : iovs ) {
+	std::shared_ptr<EcalChannelStatus> payload = Base::fetchPayload( std::get<1>(iov) );
+	if( payload.get() ){
+	  // looping over the EB channels, via the dense-index, mapped into EBDetId's
+	  if (!payload->barrelItems().size()) return false;
+	  // set to -1 for ieta 0 (no crystal)
+	  for(int iphi = 1; iphi < 361; iphi++) fillWithValue(iphi, 0, -1);
```

Comment:
here and elsewhere, try to define the magic numbers in code as a constexpr so a) their meaning is understandable, and b) the same number is not repeated many times (eg, 361, 101, 221)